c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -364,6 +365,68 @@ struct PrivatePool {
   BlockPool small_blocks;
 };
 
+std::shared_ptr<BlockState> constructBlockState(Block* block) {
+  TORCH_CHECK(
+      block->event_count == 0,
+      "Events should have synchronized when checkpointing block");
+  std::shared_ptr<BlockState> bs = std::make_shared<BlockState>();
+  bs->stream = block->stream;
+  bs->stream_uses = block->stream_uses;
+  bs->size = block->size;
+  bs->ptr = block->ptr;
+  bs->allocated = block->allocated;
+  bs->gc_count = block->gc_count;
+  return bs;
+};
+
+BlockPoolState constructBlockPoolState(
+    BlockPool* block_pool,
+    MempoolId_t pool_id,
+    const std::vector<Block*>& pool_blocks) {
+  BlockPoolState bps;
+  bps.is_small = block_pool->is_small;
+  bps.owner_id = bps.owner_id;
+
+  std::unordered_map<Block*, BlockState*> mappings;
+
+  for (Block* block : pool_blocks) {
+    if (block->pool != block_pool) {
```

Comment:
when can this happen 🤔

Reply:
I fetched all of the blocks for the PrivatePool which include both the Large and Small pools. Will add comment